src/render/painter.js

```diff
@@ -184,11 +184,14 @@ class Painter {
         if (fog) {
             const fogStart = fog.properties.get('range')[0];
             const fogEnd = fog.properties.get('range')[1];
-            // We start culling at 80% between the fog start and end,
-            // leaving a non-noticeable 2% fog opacity change threshold.
-            const fogCullDist = fogStart + (fogEnd - fogStart) * 0.8;
-
-            this.transform.fogCullDistSq = fogCullDist *  fogCullDist;
+            const fogStrength = fog.properties.get('strength');
+            // We start culling where the fog opacity function hits 98%, leaving
+            // a non-noticeable opacity change threshold. We use an arbitrary function
+            // which bounds the true answer. See: https://www.desmos.com/calculator/lw03ldsuhy
```

Comment:
👍 nice link and description

Reply:
Good point. Before landing this PR, I'll screenshot any plots and add them to this PR in case desmos… goes out of business?

Reply:
I hope never!